### ISSUE TITLE
grouped-window-list: Closing window by middle mouse button makes all panels stuck

### ISSUE THREAD

**Issue by Geryson** [linuxmint/Cinnamon#8345]
* Cinnamon version: Cinnamon 4.0.9
 * Distribution: Mint 19.1
 * Graphics hardware *and* driver used: NVIDIA Quadro K2000M / Nvidia Driver 390.77
 * 64 bit
 * [xsession-errors-copy.log](https://github.com/linuxmint/Cinnamon/files/2800331/xsession-errors-copy.log)
 ```

**Issue**

I use an automatically hiding panel with the new _grouped-window-list_ applet introduced in Mint 19.1 Tessa Cinnamon Edition. When I close any window with my middle mouse button in the list, the given panel stucks and does not auto-hide anymore. If you have more than one auto-hiding panel, hovering above them and back makes them appear and stuck also. Only restarting Cinnamon session can solve the problem for me.

The other window-closing alternatives does not do the same to the panel (using "x" button on its header / closing from window icon's right-click "Close" button on the panel / closing from the thumbnail view of the hovered window icon on the panel).

**Steps to reproduce**

1. Put the new grouped-window-list applet on an auto-hiding panel (right click on the panel -> Panel Settings -> Hide panel automatically, if it's not done before).
2. After adding it to the panel, open the applet's settings window, and make sure that the "Middle mouse button function" is set to "Close lastly focused window in the group" (or something like this... I don't know... my system is Hungarian... :stuck_out_tongue: ) 
3. Close any of your opened windows with the middle mouse button
4. Be angry... or not...

**Expected behaviour**

The clicked window is closing, and all panels continue to auto-hide.

**Other information**

Most of the time I have only one display with one window-listing panel, but I often use two more displays, and one more panel on each one with one more window lists on each one. By this, I can see on the gven screen only the windows which are on that screen. Even when these window lists are not used, the system can see them, and window list settings are available for them. The problem occures in both situations: one display / three displays.

**Comment by jaszhix** [comment]
I can't reproduce this. Does this occur if you remove any non-cinnamon applets/desklets/extensions?

**Comment by Geryson** [comment]
Sorry, this step had no effect for me, although I've phisically removed my existing applet/desklet packages. By the way, I've used the same applets/desklets in Mint 19 with no problem. Please see my update on the question, I've just found new, interesting informations...

**Comment by icarter09** [comment]
Cinnamon 4.6.7
Mint 20 64bit

I'm not able to reproduce this issue. @Geryson are you still experiencing this issue?

**Comment by Geryson** [comment]
@icarter09, yes, I can. Nowadays I don't even try to close any window like this. But now I've tried it again, and it behaves the same. I think I'm on the same Mint and Cinnamon version that you described. Like I said at the original post above, only one thing solves it: restarting Cinnamon (Ctrl+Alt+Esc).

[Here](https://termbin.com/0747d) are my current system specs.

**Comment by icarter09** [comment]
@Geryson thanks for the video. I tried setting my panels to mirror yours and still was unable to reproduce the issue. Could you post your settings file for GWL (`~/.cinnamon/configs/grouped-window-list@cinnamon.org/*.json`) Thanks.

**Comment by Geryson** [comment]
@icarter09, of course. Actually I have two .json files in this directory.

[15.json](https://pastebin.com/Dy2wqLKa)
[26.json](https://pastebin.com/hwfrqttx)

**Comment by icarter09** [comment]
@Geryson with one panel, I don't see the issue. But with two panels, I can now produce the issue just about every time I close a window with the middle button.

**Comment by Geryson** [comment]
@icarter09, thanks for the investigation!

Normally, I would have three panels (main screen top and bottom, secondary screen top). I started testing again by using only one panel, main screen's top panel - this means: changed nothing in configuration, only removed the other 2 panels. Shut down the computer, start it again... no difference...

Then I went for the pinned apps you mentioned. If I understand correctly, adding back the default pinned apps (for me they are Firefox, Terminal and Nemo) helped you with the issue, and I assume, in the .json file of the given panel it would look like this:

`"pinned-apps": {
        "type": "generic",
        "default": [
            "firefox.desktop",
            "org.gnome.Terminal.desktop",
            "nemo.desktop"
        ],
        "value": [
            "firefox.desktop",
            "org.gnome.Terminal.desktop",
            "nemo.desktop"
        ]
    }`

Well, for me it doesn't change anything. My last test was: only one panel (main screen top), both of the GWLs removed from the panels perviously, added back only one GWL for the only panel (of course, this generated new config file, and placed back the default pinned apps mentioned above). I've restarted the PC numerous times while testing one thing after another. But sadly I still can see the issue...

**Comment by icarter09** [comment]
Thanks for the heads up. I think just removing GWL and adding it back should be enough to test against, instead of having to restart the computer each time. I'll dig some more into the matter as well.
